### DISCUSSION TITLE
Configuration of managed mode AP without using the UI doesn't work

### DISCUSSION THREAD

**Discussion by FIAV1** [RaspAP/raspap-webgui#859] [Q&A] [answered]
Hi, I tried to setup managed mode AP without using the UI following this guide: [Configuration of managed mode AP without using the UI](https://docs.raspap.com/faq/#can-i-configure-a-managed-mode-ap-without-using-the-ui), but it doesn't work.

I start from a fresh Raspberry Pi OS image, with all prerequisites met. Then I follow these steps:
- Create `/etc/raspap/hostapd.ini` file and insert configurations:
  ```bash
  sudo tee /etc/raspap/hostapd.ini << EOF
  WifiInterface = uap0
  LogEnable = 0
  WifiAPEnable = 1
  BridgedEnable = 0
  WifiManaged = wlan0
  EOF
  ````
- Restart raspapd:
  ```bash
  sudo systemctl restart raspapd.service
  ```

At this point the Raspberry just spawn an AP (so the configurations made aren't applied).

What am I missing?

**Comment by billz** [comment, marked as answer]
Per the FAQ, this only works with a valid wpa_supplicant.conf

**Reply by FIAV1** [reply]
> I start from a fresh Raspberry Pi OS image, with all prerequisites met.

In the prerequisites, I supply also a valid `wpa_supplicant.conf`. More precisely, these are the steps I followed during prerequisites setup:
- At the root of SD card, I added
  - An empty `ssh` file
  - A `wpa_supplicant.conf` file with configuration for my home network
- `sudo apt update && sudo apt full-upgrade && sudo reboot`

After reboot, the Raspberry can connect to internet and is reachable through ssh.

**Reply by billz** [reply]
Possibly stating the obvious, but you will also need a valid `hostapd.conf` with `interface=uap0`. 
You can either create this config yourself or let RaspAP do it for you.

**Reply by FIAV1** [reply]
Ok, I'll add that too. Are there other things to do? Like, I saw that some iptables rules are added too, also dnsmasq needs an update. Could you please give me a list of all the things I need to do without using the web interface?

**Reply by billz** [reply]
> Are there other things to do? 

Read the manual installation steps.

**Reply by FIAV1** [reply]
Yep, I'm currently following this way.
I'm trying to automate some basic configurations with Ansible, probably the manual installation is gonna work.
I think this should be highlighted in the FAQ, since the current answer is misleading...

**Reply by billz** [reply]
This FAQ addresses a very specific use case (setting the AP-STA option without the UI). It's not intended to be a comprehensive how-to. I've added a link to the hostapd.conf example for additional clarity. 

Beyond this, you are in uncharted territory ;)

**Reply by FIAV1** [reply]
I wrote an Ansible playbook, that follows RaspAP manual installation and by editing some configurations before copying files enables by default the AP-STA mode. Thanks for your help!

**Reply by billz** [reply]
Great! Any plans to open source it?

**Reply by FIAV1** [reply]
Would be a pleasure for me! What should I do? The playbook has quite a lot of files, should I create a side project?
Btw, an Ansible playbook could easily substitute the quick installer, and it should be more easily mantainable!

**Reply by billz** [reply]
You're welcome to add the playbook to your own repo, or `RaspAP/raspap-ansible` if you prefer. We could help promote the latter under the RaspAP org. I have a hunch that users who aren't familiar with Ansible will go for the quick installer, but you never know ;)

**Reply by FIAV1** [reply]
I did the first commit, and I pushed it (force of habit) in `main` branch

**Reply by billz** [reply]
Got it. I've updated the repro to use `main` as the default branch.
